File: 1710_Maximum Units on a Truck/MaximumUnitsonaTruck.cs
public class Solution {
    public int MaximumUnits(int[][] boxTypes, int truckSize) {
        
        Array.Sort(boxTypes, (a, b) => { return b[1].CompareTo(a[1]); });
        
        var maxUnits = 0;
        for(int i=0;i<boxTypes.Length;i++)
        {
            var boxType = boxTypes[i];
            if(boxType[0] <= truckSize)
            {
                maxUnits += (boxType[0] * boxType[1]);
                truckSize -= boxType[0];
            }else{
                maxUnits += (truckSize * boxType[1]);
                truckSize = 0;
            }
            
            if(truckSize == 0) break;
        }
        
        return maxUnits;
    }
}

File: 1710_Maximum Units on a Truck/MaximumUnitsonaTruck.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes, (a, b) -> Integer.compare(b[1], a[1]));

        int ans = 0;
        for(int[] boxType : boxTypes) {
            if(truckSize <= 0) break;
            int boxes = Math.min(truckSize, boxType[0]);
            ans += boxes * boxType[1];
            truckSize -= boxes;
        }
        return ans;
    }
}